components/servo/gpio/servo.go

```diff
@@ -313,28 +328,44 @@ func (s *servoGPIO) findPWMResolution(ctx context.Context) error {
 func (s *servoGPIO) Move(ctx context.Context, ang uint32, extra map[string]interface{}) error {
 	ctx, done := s.opMgr.New(ctx)
 	defer done()
+
 	angle := float64(ang)
+
 	if angle < s.minDeg {
 		angle = s.minDeg
 	}
 	if angle > s.maxDeg {
 		angle = s.maxDeg
 	}
+
 	pct := mapDegToDutyCylePct(s.minUs, s.maxUs, s.minDeg, s.maxDeg, angle, s.frequency)
 	if s.pwmRes != 0 {
 		realTick := math.Round(pct * float64(s.pwmRes))
 		pct = realTick / float64(s.pwmRes)
 	}
+
 	if err := s.pin.SetPWM(ctx, pct, nil); err != nil {
 		return errors.Wrap(err, "couldn't move the servo")
 	}
+
 	s.currPct = pct
 	return nil
 }
 
 // Position returns the current set angle (degrees) of the servo.
 func (s *servoGPIO) Position(ctx context.Context, extra map[string]interface{}) (uint32, error) {
 	pct, err := s.pin.PWM(ctx, nil)
+
+	// Since Stop() sets the dutyCycle to 0.0 in order to maintain the position of the servo,
+	// we are setting the dutyCycle back to the last known dutyCycle to prevent the servo
+	// from going back to zero position.
+	if pct == 0 {
+		pct = s.currPct
+		err := s.pin.SetPWM(ctx, pct, extra)
+		if err != nil {
+			return 0, errors.Wrap(err, "couldn't get servo pin duty cycle")
+		}
+	}
 	if err != nil {
```

Comment:
I think this error got moved too far down, should be below line 357.

Reply:
right! coming up